sklearn/cross_validation.py

```diff
@@ -520,11 +520,11 @@ def __init__(self, y, n_folds=3, shuffle=False,
         label_counts = bincount(y_inversed)
         min_labels = np.min(label_counts)
         if self.n_folds > min_labels:
-            warnings.warn(("The least populated class in y has only %d"
-                           " members, which is too few. The minimum"
-                           " number of labels for any class cannot"
-                           " be less than n_folds=%d."
-                           % (min_labels, self.n_folds)), Warning)
+            raise ValueError("The least populated class in y has only %d"
+                             " members, which is too few. The minimum"
+                             " number of labels for any class cannot"
+                             " be less than n_folds=%d."
+                             % (min_labels, self.n_folds))
```

Comment:
It should be `np.all(self.n_folds > label_counts)`. It is fine if the least populated class has fewer than n_labels. It is a problem if all the classes have lesser than n_labels.